backend/functions/blocks.php

```diff
@@ -91,11 +91,15 @@ function( string $block_content, array $block ) {
  */
 function lean_add_block_props( string $block_content, array $block ) : string {
 	$block_name = $block['blockName'];
+	$close_pos  = strpos( $block_content, '>' );
 
-	$close_pos     = strpos( $block_content, '>' );
-	$data_type_pos = strpos( $block_content, 'data-type' );
-	if ( false !== $close_pos && false === $data_type_pos ) {
```

Comment:
I think you could have simply kept it as was and changed the if as follows:

`if ( false !== $close_pos && ( false === $data_type_pos || $data_type_pos > $close_pos )   ) {`

Reply:
I initially considered that solution, but I think it's more performant to search for the `data-type` attribute just in the opening tag of the block instead of the whole block content (e.g. if the block had a lot of content and just at the end an element with this data attribute).